### ISSUE TITLE
Unstable release tarballs

### ISSUE THREAD

**Issue by philhofer** [facebook/zstd#2343]
At some point this summer, the base64-encoded blake2b-256 checksum of the v1.4.5.tar.gz release tarball was
```
d9Na3jBddIFbIronfi6bQz9FAUPgPZSW5HTaxZlNrlg=
```
and now it is
```
rcZLRebQ1xrh7UhXvXMs9dFwQOy3Fq9MgE8KkpCZs_k=
```

If you have a v1.4.5 release tarball from earlier this summer, you
ought to be able to reproduce this issue just by re-downloading
the release from `https://github.com/facebook/zstd/archive/v1.4.5.tar.gz`
and observing that it has changed.

There's an old copy of the tarball [here](https://b2cdn.sunfi.sh/file/pub-cdn/d9Na3jBddIFbIronfi6bQz9FAUPgPZSW5HTaxZlNrlg=) in my source CDN.

I caught this issue as part of a reproducible build that failed because
it expects source tarballs to be stable (and Github ostensibly provides stable source tarballs).

Related to issue #1040.

**Comment by philhofer** [comment]
FYI, based on my initial triage here it looks like the difference is in the gzip header, and not in the underlying tar archive. (The old gzip header had a timestamp in it; the new one does not.)

**Comment by terrelln** [comment]
Hi @philhofer,

As a solution to #1040 we started uploading our own, stable, release tarballs every release. We upload our own because the GitHub tarballs aren't guaranteed to never change. We also upload the sha256sum.

For zstd-1.4.5 we provide `zstd-1.4.5.tar.{zst,gz}` and `zstd-1.4.5.tar.{zst,gz}.sha256` on the release page https://github.com/facebook/zstd/releases/tag/v1.4.5

**Comment by philhofer** [comment]
Right, which is why I'm filing this issue: the manually-uploaded
tarball for v1.4.5 was uploaded more than once (probably?) because I
got something different just now than I got in June, despite using
exactly the same tarball URL. You'll notice the gzip timestamp on the
original v1.4.5 tarball that I linked in my issue description has a
May 22, 2020 timestamp, and the one currently available has no gzip
header timestamp. And I'm certain that both were downloaded from
exactly the same github release URL, because it's precisely the same
program that I'm executing now to fetch and build the code that's
warning me about unstable source tarballs.

In short, issue #1040 is not fixed, either because someone is manually
re-uploaded releases, or because something automated is doing the
same. Ideally we'd track down what happened here.

**Comment by Cyan4973** [comment]
`https://github.com/facebook/zstd/archive/v1.4.5.tar.gz` is the automated github release tarball

**Comment by philhofer** [comment]
Ah thanks @Cyan4973 you're correct. Didn't realize the link on the "downloads" browser page was distinct from the Github link. Sorry for the noise. Closing this issue now.
